raft/src/main/java/org/apache/kafka/raft/QuorumState.java

```diff
@@ -48,16 +48,29 @@ public QuorumState(int localId,
         this.log = logContext.logger(QuorumState.class);
     }
 
-    public void initialize(OffsetAndEpoch offsetAndEpoch) throws IOException {
+    public void initialize(OffsetAndEpoch logEndOffsetAndEpoch) throws IOException {
         // We initialize in whatever state we were in on shutdown. If we were a leader
         // or candidate, probably an election was held, but we will find out about it
         // when we send Vote or BeginEpoch requests.
-        ElectionState election = store.readElectionState();
-        if (election.epoch < offsetAndEpoch.epoch) {
+
+        ElectionState election;
+        try {
+            election = store.readElectionState();
+        } catch (final IOException e) {
```

Comment:
I think we should leave this for a follow-up so that we can get this checked in, but one problem here is distinguishing between real IO errors and missing/corrupt data. In the case of a real IO error (e.g. say we've reached the open file limit), then we'd probably want to propagate that and shutdown the broker. One way we could handle this is to let `QuorumStateStore` return `Optional<ElectionState>`. Anyway, I'll file a follow-up jira.

Reply:
Sounds good, if it's indeed an IO issue, we would also be very likely to hit it for subsequent state store write operation IIUC.